### PR TITLE
go-judge: init at 1.8.0

### PR DESCRIPTION
**Pull request by criyle**
## Description of changes

`go-judge` is a high performance sandbox server based on container technologies

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26404,6 +26404,8 @@ with pkgs;
 
   go-cqhttp = callPackage ../servers/go-cqhttp { };
 
+  go-judge = callPackage ../servers/go-judge { };
+
```
Comment (Lurkki14) [inline pkgs/top-level/all-packages.nix:26408]: New packages are recommended to be placed in `pkgs/by-name` https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md
```suggestion
```

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -3692,6 +3692,12 @@
     githubId = 1222362;
     name = "Matías Lang";
   };
+  criyle = {
+    email = "i+nixos@goj.ac";
+    name = "Yang Gao";
+    githubId = 6821729;
+    github = "criyle";
+  };
```
Comment (Lurkki14) [inline maintainers/maintainer-list.nix:3700]: Adding yourself to maintainers needs to be a separate commit; `maintainers: add criyle`

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/servers/go-judge/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  version = "1.8.0";
+
+  src = fetchFromGitHub {
+    owner = "criyle";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-iKSOD/jh7NgGUNeQxFqlZDcctUXBDC1Tjxsm0Q2iZ3I=";
+  };
+
+  vendorSha256 = "sha256-GVsRflqqt+PwVGWaNGMH4prKQ5pWqPRlsTBJZtC+7zo=";
```
Comment (kirillrdy) [inline pkgs/servers/go-judge/default.nix:17]:
```suggestion
  vendorHash = "sha256-GVsRflqqt+PwVGWaNGMH4prKQ5pWqPRlsTBJZtC+7zo=";
```

Comment (criyle) [reply pkgs/servers/go-judge/default.nix]: Thanks, updated

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (criyle) [reply maintainers/maintainer-list.nix:3700]: Thanks, moved to a separate commit

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (criyle) [reply pkgs/top-level/all-packages.nix]: Thanks, moved to `pkgs/by-name`

File: pkgs/by-name/go/go-judge/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  preBuild = ''
+    echo v${version} > ./cmd/go-judge/version/version.txt
+  '';
+
+  CGO_ENABLED = 0;
+
+  meta = with lib; {
+    description = "High performance sandbox service based on container technologies";
+    homepage = "https://github.com/criyle/go-judge";
+    license = licenses.mit;
```
Comment (Lurkki14) [inline pkgs/by-name/go/go-judge/package.nix:32]:
```suggestion
    license = licenses.mit;
    mainProgram = "go-judge";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/go/go-judge/package.nix (hunk 4 above)
Comment (criyle) [reply pkgs/by-name/go/go-judge/package.nix:32]: Thanks, updated

**Review by Lurkki14 — APPROVED** [review]
